### ISSUE TITLE
Test success percentage not displaying correctly

### ISSUE THREAD

**Issue by BaronKhan** [m8pple/arch2-2015-cw1#5]
In mips_test_framework.cpp, the percentage display for the tests does not work for me as it always displays -0.0% for everything (I am using Windows 7 with the mingw32 g++ compiler and linker). I think it is related to the %5.1lf specifier. I changed this to %.5.1f and it seems to work.

**Comment by m8pple** [comment]
I was under the impression that %lf was the way of printing doubles (rather than floats),
but it turns out I have been wrong for decades:
http://pubs.opengroup.org/onlinepubs/009695399/functions/fprintf.html

printf is a variadic function and all float arguments will be promoted to doubles
(which I knew at some level), but didn't think about why I was slavishly
writing "%f" or %lf" for float and double.

And it turns out, in principle, "%lf" is not defined, and I should be using "%f". Apparently
mingw32 actually cares.

I think I picked up the habit from scanf, where "%f" and "%lf" _do_ have a different
meaning - I have the bad habit of assuming one is the inverse of the other.

Good, I learned something today - thanks! I shall update the source file.
